### ISSUE TITLE
Test connection and Update connection button functionality unclear

### ISSUE THREAD

**Issue by ubdbra001** [twVolc/PyCamPermanent#234]
I was playing around with these for FTP transfer settings and realised that they don't do exactly what I'd expect them to.

Specifically:
"Test connection" will only try connecting to a supplied address only if the connection isn't already open. If a connection is already open then it will test that connection, even if you have changed the value in the 'IP Address' box.

**Comment by twVolc** [comment]
Yes this bit was definitely a bit of an aftrethought, mainly for debugging I'd say. It could definitely be tidied up - I guess if the IP address has changed really Test Connnection should apply that update then test it?

**Comment by tpering** [comment]
I think this is something we should address as part of the ongoing project.
Could connecting to the cameras be simpler? Is there a way of having an IP
address stored in a config (or settable in the GUI) which is then tested
and automatically connected to as needed. I.e., could we automate the
"connect" sequence somewhat?

**Comment by twVolc** [comment]
Yes I think there's definitely improvements that can be made. I think we'll always need a connect button of some sort as we need to be able to open the GUI without a connection (i.e. post processing) and there might be times during manual acquisition where you need to start pycam through the GUI first, then connect once it's started. That process maybe could be bundled into a single button that starts the software then keeps checking for the connection to become available, but for other times in the field with automated capture you'd want the opposite where you don't need to run pycam but do need to connect to the instrument. I think the alternative is having a thread constantly running in the background that is trying to connect to an instrument, but this method would add extra complexity (more threads continuously running) so presumably would have some performance impact for times when no connection is needed (i.e. post-processing).
